mobile-app/app/screens/AppNavigator/screens/Balances/components/AddressControlScreen.tsx

```diff
@@ -92,6 +94,10 @@ export function AddressControlCard ({ onClose }: { onClose: () => void }): JSX.E
       addresses.push(address)
     }
     setAvailableAddresses(addresses)
```

Comment:
Is this an expensive op? Maybe we don't need to wait for this. just call it every block but not necessarily wait for the promise to resolve

Reply:
fetchAddresses is getting call only when there is change in `wallet` and `addressLength` , If its related to `isNextAddressUsable` then we can remove it from here but In this case suppose `addressLength` get change then user have to wait for some time (wait for next block to come) for enabling add new address